insights/parsers/installed_rpms.py

```diff
@@ -594,6 +594,11 @@ def __ge__(self, other):
 
     def __le__(self, other):
         return isinstance(other, InstalledRpm) and not other.__lt__(self)
+    
+    def __hash__(self):
+        # Python 3 requires hash implementation to have hashable object.
+        arch = getattr(self, 'arch', 'unknown')
```

Comment:
> @matysek I don't believe this works as expected. Since `arch` is a defined attribute of `InstalledRpm`, `getattr` will just return the value of `self.arch`, which defaults to `None`. Did you mean to do this instead `arch = self.arch if self.arch is not None else 'unknown'`?

Hi @bfahr yes, you are right, It doesn't work as expected and we wanted to work the way you mentioned.

There is an issue that `self.arch` defaults to `None`. We agreed that for hash to be unique NVR is not enough and we need also ARCH (e.g. i686 rpms installed on x86_64). Originally, we tried to use `self.nvra`. However, our tests failed because `self.nvra` does not work with `self.arch == None`. We got exceptions like this:
```
traceback (most recent call last):
  File "/home/martin/Projects/insights-core/insights/parsers/installed_rpms.py", line 602, in __hash__
    return hash(self.nvra)
  File "/home/martin/Projects/insights-core/insights/parsers/installed_rpms.py", line 505, in nvra
    return ".".join([self.package, self.arch])
TypeError: sequence item 1: expected str instance, NoneType found
``` 

That is the reason why we used a woraround with arch. Another implementation could look like this to avoid issue with getattr:
```python
        try:
            value = self.nvra
        except TypeError:
            value = self.nvr
        return hash(value)
```
@bfahr what do you think?

Reply:
I changed implementation this way and added some tests as requested.